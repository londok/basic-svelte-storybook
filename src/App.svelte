<script>
  import Button from "./components/button.svelte";
  export let data = {
    title: "Button",
    type: null,
    click: null,
    disabled: false,
    loading: false,
  };
</script>

<p style="text-align: center;">This is Button Page</p>
<div style="align: center" data-testid="app-tid">
  <Button {data} />
</div>
