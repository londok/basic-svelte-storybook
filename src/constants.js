export const type = {
  primary: "primary",
  secondary: "secondary",
  transparent: "transparent",
};