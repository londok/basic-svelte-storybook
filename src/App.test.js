import { render, waitFor } from '@testing-library/svelte';
import App from './App.svelte';
import { type } from "./constants"
const buttonData = {
  title: "Button",
  type: null,
  click: null,
  disabled: false,
  loading: false,
};
describe('App', () => {
  it('default', async () => {
    expect.hasAssertions();
    const { container } = render(App, { props: { data: buttonData } });
    await waitFor(() => {
      expect(container.firstChild.children[0]).toHaveTextContent("This is Button Page");
      expect(container.firstChild.children[1].firstChild.classList.contains(type.primary)).toBe(true);
      expect(container).toMatchSnapshot(); // capture snapshot
    })
  });
});