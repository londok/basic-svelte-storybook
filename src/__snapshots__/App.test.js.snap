// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App default 1`] = `
<body>
  <div>
    <p
      style="text-align: center;"
    >
      This is Button Page
    </p>
     
    <div
      data-testid="app-tid"
    >
      <button
        class="button primary svelte-1p69ywv"
        data-testid="button-tid"
      >
        Button
      </button>
    </div>
  </div>
</body>
`;
